pkg/apis/application/v1alpha1/cluster_constants.go

```diff
@@ -32,26 +46,30 @@ var (
 
 	// K8sMaxIdleConnections controls the number of max idle connections in K8s REST client HTTP transport
 	K8sMaxIdleConnections = 500
+
+	// K8sTLSHandshakeTimeout defines the maximum duration to wait for a TLS handshake to complete
+	K8sTLSHandshakeTimeout = 10 * time.Second
```

Comment:
We could inline the init method. It makes it a bit easier to understand what default it.

```
K8sTLSHandshakeTimeout = env.ParseDurationFromEnv(EnvK8sTLSHandshakeTimeout, 10 * time.Second, 0, math.MaxInt32)
```

Totally nice to have. Feel free to ignore suggestion